pkg/controller/service/servicecontroller.go

```diff
@@ -767,3 +737,64 @@ func (s *cachedService) nextRetryDelay() time.Duration {
 func (s *cachedService) resetRetryDelay() {
 	s.lastRetryDelay = time.Duration(0)
 }
+
+
+// syncService will sync the Service with the given key if it has had its expectations fulfilled,
+// meaning it did not expect to see any more of its pods created or deleted. This function is not meant to be
+// invoked concurrently with the same key.
+func (sc *ServiceController) syncService(key string) error {
+	startTime := time.Now()
+	defer func() {
+		glog.V(4).Infof("Finished syncing service %q (%v)", key, time.Now().Sub(startTime))
+	}()
+	//obj shows the latest service info from apiserver
+	obj, exists, err := sc.serviceStore.Store.GetByKey(key)
+	if !exists {
+		glog.Infof("Service has been deleted %v", key)
+		return nil
+	}
+	if err != nil {
+		glog.Infof("Unable to retrieve service %v from store: %v", key, err)
+		sc.queue.Add(key)
+		return err
+	}
+
+	deltaService, ok := obj.(*api.Service)
+
+	var namespacedName types.NamespacedName
+	var cachedService *cachedService
+	if !ok {
+		key, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			return fmt.Errorf("object contained wasn't a service or a deleted key: %+v", obj)
+		}
+		cachedService, ok = sc.cache.get(key.Key)
+		if !ok {
+			return fmt.Errorf("Service %s not in cache even though the watcher thought it was. Ignoring the deletion.", key)
+		}
+		deltaService = cachedService.lastState
+		obj = deltaService
+		namespacedName = types.NamespacedName{Namespace: deltaService.Namespace, Name: deltaService.Name}
+	} else {
+		namespacedName.Namespace = deltaService.Namespace
+		namespacedName.Name = deltaService.Name
+		cachedService = sc.cache.getOrCreate(namespacedName.String())
+	}
+
+	err, retryDelay := sc.processDelta(cachedService, deltaService, namespacedName)
```

Comment:
can you add a comment on why we really need a delta? on every service update can we just get the latest service from the store, and verify that it has the expected loadbalancers? that would make this controller level triggered.